scripts/pcm-dpc/covid-19/preprocess.py

```diff
@@ -0,0 +1,193 @@
+                 'casi_da_sospetto_diagnostico': 'PositiveCasesFromClinicActivity',
+                 'casi_da_screening': 'PositiveCasesFromSurveyAndTest'}
+        for col in self.data.columns:
+            assert col in it2en
+        self.data = self.data.rename(columns=it2en)
+
+    def set_location(self):
+        raise NotImplementedError
+
+    def geo_template(self):
+        raise NotImplementedError
+
+STAT_VARS = ['Count_MedicalConditionIncident_COVID_19_'
+             'PatientHospitalizedWithSymptoms',
+             'Count_MedicalConditionIncident_COVID_19_PatientInICU',
+             'Count_MedicalConditionIncident_COVID_19_'
+             'PatientHospitalized',
+             'Count_MedicalConditionIncident_COVID_19_'
+             'PatientInHomeIsolation',
+             'Count_MedicalConditionIncident_COVID_19_ActiveCase',
+             'IncrementalCount_MedicalConditionIncident_COVID_19_'
+             'PositiveCase',
+             'IncrementalCount_MedicalConditionIncident_COVID_19_'
+             'ActiveCase',
+             'CumulativeCount_MedicalConditionIncident_COVID_19_'
+             'PatientRecovered',
+             'CumulativeCount_MedicalConditionIncident_COVID_19'
+             '_PatientDeceased',
+             'CumulativeCount_MedicalConditionIncident_'
+             'COVID_19_PositiveCase',
+             'CumulativeCount_MedicalTest_COVID_19',
+             'CumulativeCount_Person_COVID_19_Tested']
+
+class PcmDpcNational(PcmDpc):
+    """subclass processing national data"""
+    def __init__(self):
+        self.csvpath = ('https://raw.githubusercontent.com/pcm-dpc/COVID-19/'
+                        'master/dati-andamento-nazionale/dpc-covid19-ita-and'
+                        'amento-nazionale.csv')
+        self.name = "dpc-covid19-ita-national-trend"
+        self.stat_vars = STAT_VARS
+
+    def set_location(self):
+        assert (self.data['State'] == 'ITA').all()
+        self.data = self.data.drop(columns=['State'])
+
+    def geo_template(self):
+        return 'dcid:country/ITA'
+
+
+class PcmDpcRegions(PcmDpc):
+    """subclass processing regional data"""
+    def __init__(self):
+        self.csvpath = ('https://raw.githubusercontent.com/pcm-dpc/COVID-19/'
+                        'master/dati-regioni/dpc-covid19-ita-regioni.csv')
+        self.name = "dpc-covid19-ita-regional"
+        self.stat_vars = STAT_VARS
+
+    def set_location(self):
+        region_code_path = ('https://raw.githubusercontent.com/qlj-lijuan/data/'
+                            'master/scripts/istat/geos/cleaned/ISTAT_region.csv')
+        region_code = pd.read_csv(region_code_path)[['Region Code', 'NUTS2']]
+        code_dict = region_code.set_index('Region Code').to_dict()['NUTS2']
+        # region code 21 and 22 is missing from the dict above, add manually here.
+        code_dict[21] = 'nuts/ITH1'
+        code_dict[22] = 'nuts/ITH2'
+        self.data['Location'] = self.data['RegionCode'].map(code_dict)
```

Comment:
I think a list of regions is being loaded from a file.  It would be better to wrap that file in a Python class or function and call that code.

We cannot depend on anything in your repository.

Reply:
I see. I downloaded the file to this directory.